addons/medical_damage/functions/fnc_getTypeOfDamage.sqf

```diff
@@ -17,24 +17,18 @@
 
 params ["_typeOfProjectile"];
 
-// --- projectiles
-if (_typeOfProjectile isKindOf "BulletBase") exitWith {"bullet"};
-if (_typeOfProjectile isKindOf "ShotgunBase") exitWith {"bullet"};
-if (_typeOfProjectile isKindOf "GrenadeCore") exitWith {"grenade"};
-if (_typeOfProjectile isKindOf "TimeBombCore") exitWith {"explosive"};
-if (_typeOfProjectile isKindOf "MineCore") exitWith {"explosive"};
-if (_typeOfProjectile isKindOf "FuelExplosion") exitWith {"explosive"};
-if (_typeOfProjectile isKindOf "ShellBase") exitWith {"shell"};
-if (_typeOfProjectile isKindOf "RocketBase") exitWith {"explosive"};
-if (_typeOfProjectile isKindOf "MissileBase") exitWith {"explosive"};
-if (_typeOfProjectile isKindOf "LaserBombCore") exitWith {"explosive"};
-if (_typeOfProjectile isKindOf "BombCore") exitWith {"explosive"};
-if (_typeOfProjectile isKindOf "Grenade") exitWith {"grenade"};
+private _damageType = GVAR(damageTypeCache) getVariable _typeOfProjectile;
 
-// --- non-projectiles reported by custom handleDamge wrapper
-if ((_typeOfProjectile select [0,1]) isEqualTo "#") then {
-    _typeOfProjectile = _typeOfProjectile select [1];
+if (isNil "_damageType") then {
+    _damageType = switch (true) do {
+        // -- non-projectiles reported by custom handleDamage wrapper
+        case ((_typeOfProjectile select [0,1]) isEqualTo "#"): { _typeOfProjectile select [1] };
+        // -- projectiles
+        case (isText (configFile >> "CfgAmmo" >> _typeOfProjectile >> "ACE_damageType")): { getText (configFile >> "CfgAmmo" >> _typeOfProjectile >> "ACE_damageType") };
+        default {toLower _typeOfProjectile};
```

Comment:
I can't think of a good reason for this code to be reached, so a warning might help.
It's cached so should only trigger once
```suggestion
        default {
            WARNING_1("Damage Type [%1] has no ACE_damageType",_typeOfProjectile);
            toLower _typeOfProjectile
        };
```